Source/ASDisplayNode.mm

```diff
@@ -2005,8 +2033,10 @@ + (void)scheduleNodeForRecursiveDisplay:(ASDisplayNode *)node
 }
 
 /// Helper method to summarize whether or not the node run through the display process
-- (BOOL)__implementsDisplay
+- (BOOL)_implementsDisplay
 {
+  ASDN::MutexLocker l(__instanceLock__);
```

Comment:
Idea: Can we make these flags (individual flag, not the whole _flags struct) atomic and avoid locking entirely?